### ISSUE TITLE
'root-relative' urls are not resolved correctly

### ISSUE THREAD

**Issue by aantonw** [YashdalfTheGray/auto-ngtemplate-loader#5]
My templateURL looks like this:
``` templateUrl: '/src/app/tpl/home.html'```

then, I got this when i run webpack:
```ERROR in ./src/app/routes.js
Module not found: Error: Cannot resolve 'file' or 'directory' .//src/app/tpl/home.html in /app/src/app
```

**Comment by YashdalfTheGray** [comment]
@aantonw I must have gotten the email overnight and I saw this as I was headed to work. I'll look at this tonight because your use case should be supported. Sorry for the late reply.

**Comment by YashdalfTheGray** [comment]
Just as an update, I had something much more pressing to deal with the last couple of days but I've started working on this. I'm going to get some sleep now but will pick this up tomorrow.

**Comment by aantonw** [comment]
Sure, take your time. Sorry couldn't help much because of my ES6 is very limited.

**Comment by YashdalfTheGray** [comment]
@aantonw I just pushed a possible solution to `master`. If you could validate that it's something that can work for you, I'll be happy to push it to NPM tomorrow.

You can pull down an NPM package straight from Github by running the following command.
```shell
# omit --no-save for npm versions older than v5.x
npm install --no-save github:YashdalfTheGray/auto-ngtemplate-loader
```
It should let you specify a `pathResolver` function in the options for the loader.

Documentation on the `pathResolver` function including the function signature is [here](https://github.com/YashdalfTheGray/auto-ngtemplate-loader#options) and there is an example in the examples folder called [`05-absolute-paths`](https://github.com/YashdalfTheGray/auto-ngtemplate-loader/tree/master/examples/05-absolute-paths) with a [sample `pathResolver`](https://github.com/YashdalfTheGray/auto-ngtemplate-loader/blob/master/examples/05-absolute-paths/webpack.config.js#L20) function.

**Comment by aantonw** [comment]
@YashdalfTheGray unfortunately I use webpack 1 for time being. I can't use loader options.
Since `pathResolver` should be a function, I can't pass it via loader query. 

As a reference `html-loader` use [root](https://github.com/webpack-contrib/html-loader#root-relative-urls) in loader query to solve this and its a string.

**Comment by YashdalfTheGray** [comment]
@aantonw Oh okay. I have another idea. I'm stealing this idea from the pre-v2 version of `sass-loader` where I can have an option that you can pass into the loader called `useResolverFromConfig` which you can set to `true` and add a key called `autoNgTemplateLoader` to your `webpack.config.js` that has the `pathResolver` function. There is an example below.

```javascript
module.exports = {
  autoNgTemplateLoader: {
    pathResolver: p => p.replace(/src/, '..').substring(1)
  },
  module: {
    loaders: [
      {
        test: /\.js$/,
        exclude: /node_modules/, 
        loader: 'babel-loader!auto-ngtemplate-loader?variableName=testVar&useResolverFromConfig=true'
      }
    ]
  }
};
```

This way, the solution is Webpack v1 compatible. What are your thoughts?

**Comment by YashdalfTheGray** [comment]
Also, I was going to put this stuff in a design doc of some sort and ask for your feedback but that's what I initially wanted to do - provide a way to add a `root` to the loader so that you can set the root for the root-relative paths and then pass that into the `urlToRequest` from `loader-utils`.

Here's the issue that I run into (and ran into when I was trying to work on it last night). For a root-relative path, you can provide a root for it to resolve to a relative path but the `require` call still needs a path that is relative to the JS file that includes the template. Here's an example. 

Say my template sits at `/src/html/app.tpl.html` and the file that is trying to include the template is at `/src/html/app.js`. Also, assume that I have `rootPath` as an option for `auto-ngtemplate-loader` and I've set the root path to `./webapp`. Calling `urlToRequest('/src/html/app.tpl.html', './webapp')` returns the path `./webapp/src/html/app.tpl.html`. 

So now we've turned a root-relative path into something that Webpack can `require` except we still need to change that path to one that is relative `/src/js/app.js` which looking at the two paths, we can tell it's going to be `../html/app.tpl.html` but this problem could be arbitrarily complex because there is no set pattern that dictates the relation between the JS code and the HTML template in AngularJS and it can differ from project to project. 

This is why I took the `pathResolver` approach, the project maintainers know the mapping between the JS and the HTML so they can use the `pathResolver` to customize the behavior of `auto-ngtemplate-loader` to fit their project.

**Comment by aantonw** [comment]
@YashdalfTheGray good explanation. I totally agree with function as an option, because it more configurable than just a string, beside issue that you've been explained. Also using `useResolverFromConfig` as a Webpack v1 compatible solution looks good to me.

**Comment by YashdalfTheGray** [comment]
Glad you like it! I'll have this coded up soonish. I'm working on a non-risky way of getting to the webpack config object.

**Comment by YashdalfTheGray** [comment]
@aantonw alright, I just pushed the v1 workaround to master. Check it out using the command above to pull this loader straight from Github and see if that suits your needs!

**Comment by aantonw** [comment]
@YashdalfTheGray hi, this loader is now working with my codebase. Awesome!

**Comment by YashdalfTheGray** [comment]
@aantonw hey, glad the loader is working for you like you want it!

It's a little late here right now but I will cut a release for this tomorrow and put it on NPM. I'll leave the issue open until then so that I can notify.

**Comment by YashdalfTheGray** [comment]
@aantonw I just released v1.3.0 with support for path resolver in Webpack v1 on NPM. Feel free to update whenever you want. 

https://www.npmjs.com/package/auto-ngtemplate-loader
